pkg/scheduler/api/job_info.go

```diff
@@ -147,7 +148,17 @@ func getTaskID(pod *v1.Pod) TaskID {
 		return TaskID(ts)
 	}
 
-	return ""
+	return TaskID(GetDefaultTaskSpec(pod))
+}
+
+// GetDefaultTaskSpec return the pod generate name without pod index
+// if pod name is "tfjob-worker-9", return "tfjob-worker".
+func GetDefaultTaskSpec(pod *v1.Pod) string {
+	specs := strings.Split(pod.Name, "-")
```

Comment:
We support different operators, e.g. customized operator, it's hard to say what will be return after this PR :)